### PR TITLE
GitHub Action for DockerHub build and publish #19

### PR DESCRIPTION
**Pull request by jsanchez556**
Hi:

Please review this PR. I have attached the GitHub actions logs for reference and included links to the public Docker Hub repository containing the created images.

Action Logs: https://github.com/jsanchez556/stone-packaging/actions/runs/11147558391
DockerHub: https://hub.docker.com/r/blockchainercr/stone-packaging/tags

### CODE REVIEW

**Review by dmirgaleev — CHANGES_REQUESTED** [review]
Good job! May I ask for a couple changes? First, let's fix tags and image names. For your fork following images should be created:
docker pull blockchainercr/stone-prover
docker pull blockchainercr/cpu_air_prover
docker pull blockchainercr/cpu_air_verifier

Also, please, take a look at https://hub.docker.com/r/blockchainercr/stone-packaging/tags. For each image mentioned above, should be created following tags:
- latest (created when you push something to the master branch)
- v*.*.* (created, when you push a tag, for instance v1.0.0)
If you can't find a way to do that in one single workflow, you can split it in two

I think for Docker Hub we can drop image creation for PRs for now.

**Comment by jsanchez556** [comment]
> Good job! May I ask for a couple changes? First, let's fix tags and image names. For your fork following images should be created: docker pull blockchainercr/stone-prover docker pull blockchainercr/cpu_air_prover docker pull blockchainercr/cpu_air_verifier
> 
> Also, please, take a look at https://hub.docker.com/r/blockchainercr/stone-packaging/tags. For each image mentioned above, should be created following tags:
> 
> * latest (created when you push something to the master branch)
> * v*._._ (created, when you push a tag, for instance v1.0.0)
>   If you can't find a way to do that in one single workflow, you can split it in two
> 
> I think for Docker Hub we can drop image creation for PRs for now.

@dmirgaleev please check again. ;)

File: .github/workflows/docker-publish.yml [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+
+      - name: Determine Docker Tags
+        id: set-tag
+        run: |
+          if [[ "${GITHUB_REF}" == refs/tags/* ]]; then
+            TAG_NAME=$(echo ${GITHUB_REF} | sed 's/refs\/tags\///')
+            echo "DOCKER_TAGS=${TAG_NAME}" >> $GITHUB_ENV
+          elif [[ "${GITHUB_REF}" == "refs/heads/master" ]]; then
+            echo "Latest version tags..."
```
Comment (dmirgaleev) [inline .github/workflows/docker-publish.yml:52]: Great! May I ask you to trigger on-push-master to test it?

**Review by dmirgaleev — CHANGES_REQUESTED** [review]

File: .github/workflows/docker-publish.yml (hunk 1 above)
Comment (jsanchez556) [reply .github/workflows/docker-publish.yml:52]: Is there any problem with doing it on master? Or can I do it on another branch for testing?

Comment (jsanchez556) [reply .github/workflows/docker-publish.yml:52]: Done @dmirgaleev, please check.

https://github.com/jsanchez556/stone-packaging/actions/runs/11153195071/

File: build.sh
Comment (dmirgaleev) [inline build.sh:1]: Please, fix permissions for the file, if it's not necessary to have 755. Return it back to 644

**Review by dmirgaleev — CHANGES_REQUESTED** [review]
Perfect! Final adjustment:) Also, as an additional task, you can instead of master push use   
```
repository_dispatch:
    types: [prover-update]
```
But not necessary to do so right now.

**Comment by jsanchez556** [comment]
Sure @dmirgaleev, working on it.

**Review by dmirgaleev — APPROVED** [review]
Good job!

**Pull request merged**